File: LeetCode/243-shortest-word-distance/ex_243.cpp
class Solution {
public:
    int shortestDistance(vector<string>& words, string word1, string word2) {
        int p1 = -1, p2 = -1, distance = INT_MAX;
        for (int i = 0; i != words.size(); ++i) {
            if (word1 == words[i]) p1 = i;
            if (word2 == words[i]) p2 = i;
            if (p1 != -1 && p2 != -1)
                distance = min(distance, abs(p1 - p2));
        }
        return distance;
    }
};

File: LeetCode/243-shortest-word-distance/ex_243.py
class Solution(object):
    def shortestDistance(self, words, word1, word2):
        """
        :type words: List[str]
        :type word1: str
        :type word2: str
        :rtype: int
        """
        p1 = -1
        p2 = -1
        distance = sys.maxsize
        for i in xrange(len(words)):
            if word1 == words[i]:
                p1 = i
            if word2 == words[i]:
                p2 = i
            if p1 != -1 and p2 != -1:
                distance = min(distance, abs(p1 - p2))
        return distance
        